pkg/controller.v1/common/service.go

```diff
@@ -277,13 +279,6 @@ func (jc *JobController) CreateNewService(job metav1.Object, rtype apiv1.Replica
 		return err
 	}
 
-	// Convert ReplicaType to lower string.
-	expectationServicesKey := expectation.GenExpectationServicesKey(jobKey, rtype)
```

Comment:
Seems the change is to move code snippet to below? Is there a difference? line 311-313

Reply:
If we `expectCreation` here, the informer will never observe this creation since error is returned(such as `GetPortsFromJob`) before service is created.
As i mentioned in first comment(No. 2).

Reply:
I see, it does have some places return `err` before creating the services. Nice catch

Reply:
Github collapse some code snippets which make the review harder. :D